### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by DavidS**

### CODE REVIEW

File: spec/classes/ntp_spec.rb [hunk 1]
```diff
@@ -3,39 +3,30 @@
...
-          super().merge({ :osfamily => 'Suse', :operatingsystem => 'SLES', :operatingsystemmajrelease => '12' })
-        when 'Fedora'
-          super().merge({ :osfamily => 'RedHat', :operatingsystem => system, :operatingsystemmajrelease => '22' })
-        else
-          super().merge({ :osfamily => system, :operatingsystem => system})
-        end
+  on_supported_os.select { |_, f| f[:os]['family'] != 'Solaris' }.each do |os, f|
+    context "on #{os}" do
+      let(:facts) do
+        f.merge(super())
```
Comment (bmjen) [inline spec/classes/ntp_spec.rb]: You've reversed the merge operation here, does that still produce the desired output? In the previous implementation, it merged the os specific facts into `super()` and now it is merging `super()` into the os facts. This merge will overwrite any duplicate keys in `f` with the value in `super()` with this implementation... even `nil` values.

Comment (DavidS) [inline spec/classes/ntp_spec.rb]: Yes, that's intentional. Prior to this, super had defaults that had the os-specific facts added to it. Using rspec-puppet-facts, `f` has ALL facts, and super() becomes more specific, as it is specific for the current test group.

File: Gemfile [hunk 2]
```diff
@@ -20,6 +20,7 @@ group :development, :unit_tests do
   gem 'puppetlabs_spec_helper',    :require => false
   gem 'rspec-puppet', '>= 2.3.2',  :require => false
   gem 'simplecov',                 :require => false
+  gem 'rspec-puppet-facts', :require => false
```
Comment (tphoney) [inline Gemfile]: needs aligned

File: manifests/init.pp [hunk 3]
```diff
@@ -106,7 +106,7 @@
   }
 
   if $autoupdate {
-    notice('autoupdate parameter has been deprecated and replaced with package_ensure.  Set this to latest for the same behavior as autoupdate => true.')
+    notice('ntp: autoupdate parameter has been deprecated and replaced with package_ensure. Set package_ensure to latest for the same behavior as autoupdate => true.')
```
Comment (tphoney) [inline manifests/init.pp]: nice change

**Comment by tphoney** [comment]
@DavidS +1 apart from the alignment

File: spec/spec_helper_acceptance.rb [hunk 4]
```diff
@@ -46,10 +46,7 @@
   # Configure all nodes in nodeset
   c.before :suite do
     hosts.each do |host|
-      on host, "mkdir -p #{host['distmoduledir']}/ntp"
-      %w(lib manifests templates metadata.json).each do |file|
-        scp_to host, "#{proj_root}/#{file}", "#{host['distmoduledir']}/ntp"
-      end
+      copy_module_to(host, :source => proj_root, :module_name => 'ntp')
```
Comment (gregohardy) [inline spec/spec_helper_acceptance.rb]: +1

**Comment by gregohardy** [comment]
I'm happy. If no one objects. I will merge when travis has passed.

File: spec/classes/ntp_spec.rb (hunk 1 above)
Comment (bmjen) [inline spec/classes/ntp_spec.rb]: 👍  Thanks for the explanation.

**Pull request merged**